pkg/sql/sqlbase/structured.go

```diff
@@ -4470,6 +4515,17 @@ func MakeSimpleAliasTypeDescriptor(typ *types.T) *TypeDescriptor {
 	}
 }
 
+// MakeTypeDescriptor creates a type descriptor. It does not fill in kind
+// specific information about the type.
+func MakeTypeDescriptor(parentID, parentSchemaID, id ID, name string) *TypeDescriptor {
```

Comment:
In every case, you immediately dereference the returned pointer. I think this should just return a `TypeDescriptor`, since we're copying anyways.